src/app/clusters/service-area-server/service-area-server.cpp

```diff
@@ -0,0 +1,1122 @@
+
+    case Commands::SkipCurrentLocation::Id:
+        return CommandHandlerInterface::HandleCommand<Commands::SkipCurrentLocation::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleSkipCurrentLocationCmd(ctx); });
+    }
+}
+
+//*************************************************************************
+// attribute readers
+
+CHIP_ERROR Instance::ReadSupportedLocations(AttributeValueEncoder & aEncoder)
+{
+    if (mDelegate->GetNumberOfSupportedLocations() == 0)
+    {
+        return aEncoder.EncodeNull();
+    }
+
+    return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
+        uint8_t locationIndex = 0;
+        LocationStructureWrapper supportedLocation;
+
+        while (mDelegate->GetSupportedLocationByIndex(locationIndex++, supportedLocation))
+        {
+            ReturnErrorOnFailure(encoder.Encode(supportedLocation));
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+CHIP_ERROR Instance::ReadSupportedMaps(AttributeValueEncoder & aEncoder)
+{
+    if (mDelegate->GetNumberOfSupportedMaps() == 0)
+    {
+        return aEncoder.EncodeNull();
+    }
+
+    return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
+        uint32_t mapIndex = 0;
+        MapStructureWrapper supportedMap;
+
+        while (mDelegate->GetSupportedMapByIndex(mapIndex++, supportedMap))
+        {
+            ReturnErrorOnFailure(encoder.Encode(supportedMap));
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+CHIP_ERROR Instance::ReadSelectedLocations(AttributeValueEncoder & aEncoder)
+{
+    if (mDelegate->GetNumberOfSelectedLocations() == 0)
+    {
+        return aEncoder.EncodeNull();
+    }
+
+    return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
+        uint32_t locationIndex = 0;
+        uint32_t selectedLocation;
+
+        while (mDelegate->GetSelectedLocationByIndex(locationIndex++, selectedLocation))
+        {
+            ReturnErrorOnFailure(encoder.Encode(selectedLocation));
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+CHIP_ERROR Instance::ReadProgress(AttributeValueEncoder & aEncoder)
+{
+    if (mDelegate->GetNumberOfProgressElements() == 0)
+    {
+        return aEncoder.EncodeNull();
+    }
+
+    return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
+        uint32_t locationIndex = 0;
+        Structs::ProgressStruct::Type progressElement;
+
+        while (mDelegate->GetProgressElementByIndex(locationIndex++, progressElement))
+        {
+            ReturnErrorOnFailure(encoder.Encode(progressElement));
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+//*************************************************************************
+// command handlers
+
+void Instance::HandleSelectLocationsCmd(HandlerContext & ctx, const Commands::SelectLocations::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "Service Area: HandleSelectLocationsCmd");
+
+    uint32_t listIndex = 0;
```

Comment:
this seems to be used only as a parameter that is later unused ... why are we declaring this on top? Why not near `mDelegate->GetSelectedLocationByIndex`? Please reduce the scope of these variables to the minimum.